src/api/main/routing.ts

```diff
@@ -173,4 +174,116 @@ export type BestRouteResponse = {
+ * If there is a price impact issue Rango won't allow dApps to createTransaction and it will fail because of the risk of huge loss for user.
+ * @property {SwapResult[]} swaps - List of required swaps to swap X to Y with the expected outputAmount
+ * @property {SimulationScore[]} scores - List of scores calculated for each preference aspect
+ * @property {RouteTag[]} tags - List of tags attributed to each route considering every aspect
+ * @property {BlockchainValidationStatus[] | null} validationStatus - Pre-requisites check result. It will be null if
+ * the request checkPrerequisites was false
+ * @property {string[]} missingBlockchains - List of all blockchains which are necessary to be present for the best
+ * route and user has not provided any connected wallets for it. A null or empty list indicates that there is no problem.
+ * @property {boolean} walletNotSupportingFromBlockchain - A warning indicates that none of your wallets have the same
+ * blockchain as X asset
+ */
+export type MultiRouteSimulationResult = {
+  requestId: string
+  outputAmount: string
+  resultType: RoutingResultType
+  swaps: SwapResult[]
+  scores: { preferenceType: PreferenceType; score: number }[]
+  tags: RouteTag[]
+  validationStatus: BlockchainValidationStatus[] | null
+  missingBlockchains: string[] | null
+  walletNotSupportingFromBlockchain: boolean
+}
+
+/**
+ * The best route request body for multi-routing
+ */
+export type MultiRouteRequest = BestRouteRequest
+
+/**
+ * The best route response for multi-routing, if the result fields is null, it means that no route is found
+ *
+ * @property {string} routeId - The unique roteId generated for this request by server
+ * @property {MultiRouteSimulationResult} results - List of best routes data
+ */
+export type MultiRouteResponse = BestRouteResponse & {
+  routeId: string
+  results: MultiRouteSimulationResult[]
+}
+
+/**
+ * The body of confirmation request for selected route.
+ *
+ * @property {string} requestId - The unique requestId which is generated for this request by the server. It should be
+ * passed down to all other endpoints if this swap continues on. e.g. d10657ce-b13a-405c-825b-b47f8a5016ad
+ * @property {{ [key: string]: string }} selectedWallets - Map of blockchain to selected address
+ * @property {string} [destination] - Custom destination for the route
+ */
+export type ConfirmSwapRequest = {
+  requestId: string
+  selectedWallets: { [key: string]: string }
+  destination?: string
+}
+
+/**
+ * The response of confirmation request for selected route
+ *
+ * @property {string | null} error - Error message about the incident if ok == false.
+ * @property {string | null} errorCode - Error code about the incident if ok == false.
+ * @property {boolean} ok - If true, the result has a value and error message is null.
+ * @property {BestRouteResponse | null} result - result of confirm swap.
+ */
+export type ConfirmSwapResponse = {
```

Comment:
I think `ConfirmRouteRequest` and `ConfirmRouteResponse` is better as swap word is used for  our basic api.